src/Sylius/Behat/Context/Api/Admin/ManagingCountriesContext.php

```diff
@@ -0,0 +1,379 @@
+    }
+
+    /**
+     * @When I delete the :province province of this country
+     */
+    public function iDeleteTheProvinceOfThisCountry(ProvinceInterface $province): void
+    {
+        /** @var CountryInterface $country */
+        $country = $this->sharedStorage->get('country');
+
+        foreach ($country->getProvinces() as $key => $provinceValue) {
+            if ($province->getId() === $provinceValue->getId()) {
+                $this->client->removeSubResource('provinces', $key);
+            }
+        }
+    }
+
+    /**
+     * @When I remove :province province name
+     */
+    public function iRemoveProvinceName(ProvinceInterface $province)
+    {
+        /** @var CountryInterface $country */
+        $country = $this->sharedStorage->get('country');
+
+        foreach ($country->getProvinces() as $key => $provinceValue) {
+            if ($province->getId() === $provinceValue->getId()) {
+                $this->client->updateSubResource('provinces', $key, ['code' => $provinceValue->getCode(), 'name' => '']);
+            }
+        }
+    }
+
+    /**
+     * @Then province with code :provinceCode should not be added in this country
+     */
+    public function provinceWithCodeShouldNotBeAddedInThisCountry(string $provinceCode)
+    {
+        /** @var CountryInterface $country */
+        $country = $this->sharedStorage->get('country');
+
+        $response = $this->client->show($country->getCode());
+        $countryFromResponse = $this->responseChecker->getResponseContent($response);
+
+        foreach ($countryFromResponse['provinces'] as $provinceFromResponse)
+        {
+            /** @var ProvinceInterface $province */
+            $province = $this->iriConverter->getItemFromIri($provinceFromResponse);
+            Assert::false(
+                $province->getCode() === $provinceCode,
+                sprintf('The country "%s" should not have the "%s" province', $country->getName(), $province->getName())
+            );
+        }
+    }
+
+    /**
+     * @Then this country should not have the :provinceName province
+     * @Then province with name :provinceName should not be added in this country
+     */
+    public function thisCountryShouldNotHaveTheProvince(string $provinceName): void
+    {
+        /** @var CountryInterface $country */
+        $country = $this->sharedStorage->get('country');
+
+        $response = $this->client->show($country->getCode());
+        $countryFromResponse = $this->responseChecker->getResponseContent($response);
+
+        foreach ($countryFromResponse['provinces'] as $provinceFromResponse)
+        {
+            /** @var ProvinceInterface $province */
+            $province = $this->iriConverter->getItemFromIri($provinceFromResponse);
+            Assert::false(
+                $province->getName() === $provinceName,
+                sprintf('The country "%s" should not have the "%s" province', $country->getName(), $province->getName())
+            );
+        }
```

Comment:
This function body is very similar to the previous one, maybe we can extract something?

Reply:
I create function getProvincesOfCountry().